comma/model.py

```diff
@@ -0,0 +1,190 @@
+"""Model class definition
+"""
+from comma.individual import Individual
+import pandas as pd
+from . import read_json_as_dict, _get_one_hot_encoded_features, PARAMS_INDIVIDUAL, PARAMS_MODEL
+import os
+import numpy as np
+from tqdm import tqdm
+
+
+class Model:
+    _status = dict()
+    _required_params = ['size', 'steps', 'actions', 'status',
+                        'lockdown_policies', 'lockdown']
+
+    def __init__(self):
+        self.agents: list = None
+        self.simulation_id: int = None
+        self.current_step: int = 0 # keep track of the current simulation step
+        self.lockdown_status: dict = {}
+
+    def setup(self, size: int, dir_params: str, use_ipf: bool = False):
+        """Setup the model with input parameters.
+
+        Args:
+            size (int): size of the population
+            dir_params (str): dir to the folder containing hypothesis and parameter files.
+            use_ipf (bool): If true, use the populate_ipf() method to populate the agents
+                            If false, use the standard populate method.
+        """
+        self.param_file_validation(dir_params)
+        if use_ipf:
+            self.agents = Individual.populate_ipf(size, dir_params)
+        else:
+            self.agents = Individual.populate(size, dir_params)
+        return self.agents
+
+    def param_file_validation(self, dir_params: str):
+        """Validate files in the parameter folder.
+        
+        Args:
+            dir_params (str): dir to the folder containing hypothesis and parameter files.
+        """
+        # check if parameter files exist
+        fpath_params_individual = os.path.join(dir_params, PARAMS_INDIVIDUAL)
+        fpath_params_model = os.path.join(dir_params, PARAMS_MODEL)
+        assert os.path.isfile(fpath_params_individual), \
+            "Parameter file not found: %s." % PARAMS_INDIVIDUAL
+        assert os.path.isfile(fpath_params_model), \
+            "Parameter file not found: %s" % PARAMS_MODEL
```

Comment:
Use an exception to indicate an error in the user input (here: the parameters file), and `assert` only to validate the code itself.

Here, this seems redundant because the functions that read the files later should raise the respective exception.
```suggestion
```